### DISCUSSION TITLE
How do you separate request calls and response?

### DISCUSSION THREAD

**Discussion by oozman** [livewire/livewire#3614] [1. Help]
I'm trying to separate request calls but I'm getting a big response from the methods.

Ex:

```js
@this.bigFatAction(1)
@this.bigFatAction(2)
@this.bigFatAction(3)
```

Livewire will send a big response with all the data from these methods.

Is it possible to separate calls into separate HTTP request? Do you have any pointers how to achieve this?

Thanks.

**Comment by joshhanley** [comment]
@oozman can you explain a bit more about your use case and what you are hoping to achieve by splitting the requests? As generally due to Livewire re-rendering the whole component, it doesn't make sense to split the requests.

That being said, there could be some ways around it depending on the need.

Thanks!

**Comment by oozman** [comment]
hi @joshhanley hmm, I have a stat page where I show graphs and some data from a heavy data queries.

One method is processing maybe 30-60 seconds before it returns the data (due to parsing and stuff)
One method is also taking 10-15 seconds
One method is just taking 1 second.

These are in one page component.
So everytime I load the page and call every method on load, Livewire combines all these requests into 1 and returns the reponse once as well.

The problem is my server will wait for the 60 seconds query, 15 seconds and 1 second to finish, adding maybe about 76 seconds before something is rendered for the user.

My question is about, if I can fire them one by one, I should be able to render something (progressively) in the browser instead of waiting for everything to finish.


I hope this clarifies my question. Thanks!

_Note: Depending on the user's data, the longer it takes._

**Comment by joshhanley** [comment]
@oozman ah right, ok thanks, that makes sense. Personally, in that instance, I'd probably make them separate Livewire components if you can, then they each will fire their own network request to collect data.

But I think you could also put a delay of >5ms between each action call, and that would also separate them into separate requests.

See here https://github.com/livewire/livewire/blob/396c5c910ecf747a55ad4bc1f8d07790068154df/js/component/index.js#L218-L225

Hope this helps!

**Comment by eilerth** [comment]
Unfortunately, what I've experienced with livewire v3, is that even if separated into multiple components and allowing >5ms between each action call, it'll still wait for the prior call to complete before attempting the next one.

I've submitted this PR as a potential solution to the problem: https://github.com/livewire/livewire/pull/7102

Essentially, it allows you to add "isolate" to a livewire component to prevent it from bundling it's requests with other livewire components.
`<livewire:my-component isolate />`
